### PR TITLE
Calling an async api without providing a callback is deprecated

### PR DESCRIPTION
**Pull request by macdja38**
Also mixed tabs and spaces is a bit... bad... but that's an issue for another day.

### CODE REVIEW

**Comment by EricRabil** [comment]
Not that anything is wrong with the way you set up your lambdas, but you could have condensed it into a one line operation since it had one action and you don't need an if statement because the writeFile callback is only used for errors.
```
fs.writeFile(blah, e => console.error(e))
```

**Comment by EricRabil** [comment]
My bad, in that case you could also do `fs.writeFile(blah, e => e ? console.error(e) : null)`

**Comment by macdja38** [comment]
@EricRabil I think my proposal is a bit easier for newer users to the language to understand... which based on the surrounding code is ideal in this scenario... that said if you want to start playing code golf... 

    fs.writeFile(blah, e => e ? console.error(e) : 0)

The above is three characters shorter.

File: discord_bot.js [hunk 1]
```diff
@@ -53,7 +53,9 @@ Permissions.checkPermission = function (user,permission){
 	} catch(e){}
 	return false;
 }
-fs.writeFile("./permissions.json",JSON.stringify(Permissions,null,2));
+fs.writeFile("./permissions.json",JSON.stringify(Permissions,null,2), (err) => {
```
[the remark this answers is not in the document]
Comment (macdja38) [reply discord_bot.js]: That would suppress an error if one does occur. You really don't want errors writing stuff being suppressed like that.

[the remark this answers is not in the document]
Comment (DeathlyPlays) [reply discord_bot.js]: Fair enough

**Comment by macdja38** [comment]
@EricRabil Just had a better suggestion.

```
fs.writeFile(blah, e => e && console.error(e))
```

**Pull request merged**